### PR TITLE
Move Identity.Core and Identity.Stores to NetCoreApp3.0

### PR DESCRIPTION
**Pull request by Tratcher**
#3754 This is blocking DataProtection https://github.com/aspnet/AspNetCore/pull/4473

### CODE REVIEW

**Comment by Tratcher** [comment]
@natemcmaster these will also need to be re-evaluated from the shared framework perspective.

File: src/Identity/src/Core/Microsoft.Extensions.Identity.Core.csproj [hunk 1]
```diff
@@ -1,8 +1,8 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (HaoK) [inline src/Identity/src/Core/Microsoft.Extensions.Identity.Core.csproj]: Random question, why is this line showing up as a diff?

**Review by HaoK — APPROVED** [review]

File: src/Identity/src/Core/Microsoft.Extensions.Identity.Core.csproj (hunk 1 above)
Comment (natemcmaster) [reply src/Identity/src/Core/Microsoft.Extensions.Identity.Core.csproj]: The UTF8 BOM probably changed

**Review by natemcmaster — DISMISSED** [review]
Can you also remove these lines? https://github.com/aspnet/AspNetCore/blob/c612eb8730a7b8794493af374b36d8dd7480cf9a/build/SharedFrameworkOnly.props#L23-L24

**Comment by natemcmaster** [comment]
If these assemblies go to netcoreapp3.0, there doesn't appear to be any value in shipping as standalone nuget packages. I'm okay with this.

@Tratcher Once you update SharedFrameworkOnly.props to reflect this, you might also need mark this packages as 'noship' in build/artifacts.props. IIRC the build will fail if you don't.

**Comment by PinpointTownes** [comment]
I thought the `Microsoft.Extensions.*` packages were supposed to keep supporting `netstandard20`. Has anything changed?

**Comment by Tratcher** [comment]
@PinpointTownes we realized these two never should have been renamed to Extensions. See https://github.com/aspnet/AspNetCore/pull/4473#issuecomment-445347970

**Comment by PinpointTownes** [comment]
@Tratcher correct me if I'm wrong, but splitting and moving the core Identity components to `Microsoft.Extensions.*` was not an oversight but a **deliberate** change allowing to use Identity in non-ASP.NET Core scenarios (e.g in an Orleans/Service Bus/RPC-like membership service).

For me, it was an extremely positive change (not perfect tho', as namespaces, crypto dependencies or projects like `Microsoft.AspNetCore.Identity.EntityFrameworkCore` haven't been renamed to reduce the number of breaking changes).

It's a bit harsh to say _a posteriori_ that you didn't really want to do that and that the poor guys who dared use these packages outside of ASP.NET Core are once again left in a bad position.

**Comment by Tratcher** [comment]
Between the namespaces and the dependencies it was never properly factored out of AspNet. They're going to need to do additional work if they want to enable that properly.

**Comment by PinpointTownes** [comment]
> Between the namespaces and the dependencies it was never properly factored out of AspNet. They're going to need to do additional work if they want to enable that properly.

Technically, I can guarantee you Identity Core 2.0 worked like a charm outside of ASP.NET Core (I've used it in a gRPC service, in a .NET Core console without ASP.NET Core and in a .NET Framework app and I've never seen any issue).

Naming of certain namespaces or dependencies was not ideal, but it didn't prevent it from working.

**Comment by Tratcher** [comment]
@PinpointTownes take it up with @blowdart to see if he's willing to complete the refactoring. Last I checked he said no.

In the meantime I'll merge this to unblock our other components.

**Comment by blowdart** [comment]
It was never official, however if this move means the unofficial scenarios of using outside are no longer support that, and there are no longer separate packages, then it should be reconsidered as I know of a few large customer scenarios where they're using ASP.NET identity in things like WCF backends, and even WPF apps. cc @DamianEdwards

**Pull request merged**